src/markdown-pages/install-with-kurl/migrating-csi.md

```diff
@@ -3,73 +3,206 @@ path: "/docs/install-with-kurl/migrating-csi"
+* [Change the CSI Add-on in a Cluster](#change-the-csi-add-on-in-a-cluster)
+* [Troubleshoot Longhorn Data Migration](#troubleshoot-longhorn-data-migration)  
 
-When initially installing the following kURL spec:
+## Supported CSI Migrations
 
-```
-apiVersion: cluster.kurl.sh/v1beta1
-kind: Installer
-metadata:
-  name: old
-spec:
-  kubernetes:
-    version: 1.19.12
-  docker:
-    version: 20.10.5
-  weave:
-    version: 2.6.5
-  rook:
-    version: 1.0.4
-```
+_**Important**_: kURL does not support Longhorn. If you are currently using Longhorn, you must migrate data from Longhorn to either OpenEBS or Rook.
 
-You can then automatically migrate data _from_ Rook _to_ OpenEBS with Local PV using the following kURL spec. This requires OpenEBS 3.3.0 or newer. 
+This table describes the CSI add-on migration paths that kURL supports:
 
-```
-apiVersion: cluster.kurl.sh/v1beta1
-kind: Installer
-metadata:
-  name: new
-spec:
-  kubernetes:
-    version: 1.19.12
-  docker:
-    version: 20.10.5
-  weave:
-    version: 2.6.5
-  openebs:
-    version: 3.3.0
-    isLocalPVEnabled: true
-```
+| From      | To        | Notes                                                                                                                                                                                                                  |
+|-----------|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Longhorn  | OpenEBS 3.3.0 and later  | Migrating from Longhorn to OpenEBS 3.3.0 or later is recommended for single-node installations. |
+| Longhorn  | Rook 1.10.8 and later | Migrating from Longhorn to Rook 1.10.8 or later is recommended for clusters with three or more nodes where data replication and availability are requirements. Compared to OpenEBS, Rook requires more resources from your cluster, including a dedicated block device. Single-node installations of Rook are not recommended. Migrating from Longhorn to Rook is not supported for single-node clusters.  |
+| Rook      | OpenEBS 3.3.0 and later  | Migrating from Rook to OpenEBS 3.3.0 or later is strongly recommended for single-node installations, or for applications that do not require data replication. Compared to Rook, OpenEBS requires significantly fewer hardware resources from your cluster. |
+
+For more information about how to choose between Rook or OpenEBS, see [Choosing a PV Provisioner](create-installer/choosing-a-pv-provisioner).
+
+## About Changing the CSI Add-on
+
+You can change the CSI provisioner that your kURL cluster uses by updating the CSI add-on in your kURL specification file. Then, when you upgrade a kURL cluster using the new specification, the kURL installer detects the change that you made to the CSI add-on and begins automatically migrating data from the current provisioner to the new one.
```

Comment:
I moved the example YAML files that were previously at the top down to the procedure. It seemed like they might be too tempting up here for someone to just copy and paste without reading

Reply:
Excellent idea!